Add dev spec for PowerToys hotkey conflict detection feature (#5519)
MicrosoftDocs/windows-dev-docs: hub/powertoys/hotkey-conflict-detection.md

> # PowerToys Feature Specification: Hotkey Conflict Detection System

Reviewer: @shuaiyuanxx, is this end-user documentation or something that should be added to the PowerToys product repo instead of the Microsoft Learn documentation?

Reviewer: If this is Learn documentation, it needs to have a proper metadata header. H1 should be sentence case unless any part of it is a product name. Metadata titles use title case.

```suggestion
---
title: PowerToys Hotkey Conflict Detection System
description: This specification defines a comprehensive hotkey conflict detection system that provides real-time feedback and prevents conflicting hotkey configurations, enhancing the overall reliability and user experience of PowerToys.
ms.date: 7/16/2025
ms.topic: concept-article
no-loc: [PowerToys, Windows]
# Customer intent: As a Windows power user, I want to learn how PowerToys handles conflict detection and provides feedback and configuration options.
---

# PowerToys hotkey conflict detection system

```

> ## 1. Executive Summary

Reviewer: Avoid numbering in headers. Numbering should only be used when providing step-by-step instructions.

Reviewer: This first H2 isn't really needed at all. The intro paragraph following an H1 is assumed to be the summary.
```suggestion

```

PowerToys currently allows users to assign hotkeys across various modules without conflict detection mechanisms. This leads to overlapping hotkey assignments that can cause feature activation failures or ambiguous behavior. This specification defines a comprehensive hotkey conflict detection system that provides real-time feedback and prevents conflicting hotkey configurations, enhancing the overall reliability and user experience of PowerToys.

**Key Benefits:**
- Prevent hotkey conflicts between PowerToys modules and with system-reserved hotkeys
- Provide real-time feedback to users during hotkey configuration
- Offer centralized conflict management through a unified interface
- Maintain backwards compatibility with existing PowerToys installations

## 2. Requirements

> #### Core Conflict Detection

Reviewer: We should never skip from H2 to H4 in headers on Learn. This should be an H3.

- **Registration-time validation**: When a module attempts to register a hotkey, the system must validate against existing assignments and system-reserved hotkeys
- **Real-time UI feedback**: When users modify hotkeys in the Settings UI, immediate conflict checking with visual indicators
- **Cross-module awareness**: Detection of conflicts between different PowerToys modules

#### Conflict Classification
- **In-app conflicts**: Multiple PowerToys modules assigned the same hotkey combination
- **System conflicts**: Hotkeys reserved or registered by Windows or other applications
- **Granular identification**: Each conflict includes specific module names and hotkey identifiers

#### Data Management
- **Unique hotkey identification**: Every hotkey tagged with module name and descriptive identifier
- **Configuration persistence**: Conflict state maintained across application restarts
- **Backwards compatibility**: Automatic upgrade of existing configurations

## 3. Architecture Overview

### 3.1 System Initialization Workflow
<div align="center">

> <img src="../images/powertoys-shortcut-conflict/InitStageWorkFLow.png" alt="System initialization and hotkey registration flow during PowerToys startup" width="600px">

Reviewer: Use markdown syntax for image and avoid other HTML like the surrounding `<div>`.

```suggestion
![A diagram of the system initialization and hotkey registration flow during PowerToys startup.](../images/powertoys-shortcut-conflict/InitStageWorkFLow.png")
```
